packages/axelar-wasm-std/src/msg_id/starknet_field_element_event_index.rs

```diff
@@ -0,0 +1,266 @@
+use core::fmt;
+use std::fmt::Display;
+use std::str::FromStr;
+
+use cosmwasm_std::HexBinary;
+use error_stack::Report;
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde_with::DeserializeFromStr;
+use starknet_core::types::FieldElement;
+
+use super::Error;
+use crate::nonempty;
+
+#[derive(Debug, DeserializeFromStr, Clone)]
+pub struct FieldElementAndEventIndex {
+    pub tx_hash: FieldElement, // the tx hash
+    pub event_index: u64,
+}
+
+impl FieldElementAndEventIndex {
+    pub fn tx_hash_as_hex(&self) -> nonempty::String {
+        format!("0x{}", self.tx_hash_as_hex_no_prefix())
+            .try_into()
+            .expect("failed to convert tx hash to non-empty string")
+    }
+
+    pub fn tx_hash_as_hex_no_prefix(&self) -> nonempty::String {
+        HexBinary::from(self.tx_hash.to_bytes_be())
+            .to_hex()
+            .to_string()
+            .try_into()
+            .expect("failed to convert tx hash to non-empty string")
+    }
+
+    pub fn new<T: Into<FieldElement> + FromStr>(
+        tx_id: T,
+        event_index: impl Into<u64>,
+    ) -> Result<Self, Error> {
+        Ok(Self {
+            tx_hash: tx_id.into(),
+            event_index: event_index.into(),
+        })
+    }
+}
+
+// A valid field element is max 252 bits, meaning 62 or 63 hex characters after 0x.
+// We require the hex to be 64 characters, meaning that it should be padded with zeroes.
+const PATTERN: &str = "^(0x[0-9a-f]{64})-(0|[1-9][0-9]*)$";
+lazy_static! {
+    static ref REGEX: Regex = Regex::new(PATTERN).expect("invalid regex");
+}
+
+impl FromStr for FieldElementAndEventIndex {
+    type Err = Report<Error>;
+
+    fn from_str(message_id: &str) -> Result<Self, Self::Err>
+    where
+        Self: Sized,
+    {
+        // the PATTERN has exactly two capture groups, so the groups can be extracted safely
+        let (_, [tx_id, event_index]) = REGEX
+            .captures(message_id)
+            .ok_or(Error::InvalidMessageID {
+                id: message_id.to_string(),
+                expected_format: PATTERN.to_string(),
+            })?
+            .extract();
+        let tx_id_chunk = &tx_id[2..];
```

Comment:
This is also how the code is in other msg_id formats.
I'll restrain from changing the code here, because this file is touchy for axelar. I'll leave it, because it's their change. They'll also review and decide.